### PR TITLE
Rename styling using BEM.

### PR DESCRIPTION
**Pull request by Irsanarisandy**
- Rename styling using BEM and use bootstrap breakpoints.
- Vue components doesn't work when name is same as built-in tags, hence I renamed it to customnav instead.
- Move navigation styling to components folder, still within scss folder.

### CODE REVIEW

File: themes/app/src/js/app.js [hunk 1]
```diff
@@ -4,6 +4,6 @@ import CustomNav from './components/CustomNav.vue';
 new Vue({
     el: '#app',
     components: {
-        'custom-nav': CustomNav,
+        'customnav': CustomNav,
```
Comment (pjayme) [inline themes/app/src/js/app.js]: hmm.. I thought we were renaming this to `nav` ?

Comment (Irsanarisandy) [reply themes/app/src/js/app.js]: For naming of Vue component, it throws an error on the console log if I name it 'nav' since it's a built-in tag

Comment (pjayme) [reply themes/app/src/js/app.js]: oh i see, can do `main-nav`

File: themes/app/src/js/components/CustomNav.vue [hunk 2]
```diff
@@ -1,18 +1,18 @@
 <template>
-    <nav class="custom-nav">
-        <div class="top-menu">
-            <a href="/"><img id="logo"/></a>
+    <nav class="customnav">
```
Comment (pjayme) [inline themes/app/src/js/components/CustomNav.vue]: ah I thought you got what I meant when I said update to `main-nav` instead

I also meant update all the prefix classes throughout the pull request :) So at the end of the refactor we have no instances of `customnav` and they are all replaced with `main-nav` :)

Comment (Irsanarisandy) [reply themes/app/src/js/components/CustomNav.vue]: ok, will do now

**Pull request merged**